testing/python/setup_only.py

```diff
@@ -267,3 +267,39 @@ def test_arg(arg):
     result.stdout.fnmatch_lines(
         ["*SETUP    F arg*", "*test_arg (fixtures used: arg)F*", "*TEARDOWN F arg*"]
     )
+
+
+def test_setup_show_with_KeyboardInterrupt_in_test(testdir):
+    """ Verifies that setups are shown and tests are executed even if there was a KeyboardInterrupt in a test. """
+    p = testdir.makepyfile(
+        """
+        import pytest
+        @pytest.fixture
+        def arg():
+            assert True
+        def test_arg(arg):
+            raise KeyboardInterrupt()
+    """
+    )
+    result = testdir.runpytest("--setup-show", p, no_reraise_ctrlc=True)
+    assert result.ret == 2
+    result.stdout.fnmatch_lines(
+        ["*SETUP    F arg*", "*test_arg (fixtures used: arg)*", "*TEARDOWN F arg*"]
+    )
+
+
+def test_setup_show_with_KeyboardInterrupt_in_fixture(testdir):
+    """ Verifies that setups are shown and tests are executed even if there was a KeyboardInterrupt in a fixture. """
+    p = testdir.makepyfile(
+        """
+        import pytest
+        @pytest.fixture
+        def arg():
+            raise KeyboardInterrupt()
+        def test_arg(arg):
+            assert True
+    """
+    )
+    result = testdir.runpytest("--setup-show", p, no_reraise_ctrlc=True)
+    assert result.ret == 2
+    result.stdout.fnmatch_lines(["*SETUP    F arg*", "*KeyboardInterrupt*"])
```

Comment:
Thanks!  I usually find it useful to also match the "*= 1 failed/error in *".

Reply:
I don't get this result in the tests.
Here is an output example.
```
# t.py
import pytest
@pytest.fixture
def arg():
    assert True
def test_arg(arg):
    raise KeyboardInterrupt()
```
pytest --setup-show t.py
========================================================== test session starts ===========================================================
platform linux -- Python 3.6.7, pytest-3.8.1.dev2489+g2e11ea610.d20191019, py-1.8.0, pluggy-0.13.0
rootdir: /home/yocaspi/workspace/pytest, inifile: tox.ini
collected 1 item                                                                                                                         

t.py 
        SETUP    F arg
        t.py::test_arg (fixtures used: arg)
        TEARDOWN F arg

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
/home/yocaspi/workspace/pytest/t.py:6: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --full-trace)
========================================================= no tests ran in 0.31s ==========================================================

Reply:
Yeah, here it would be "*= no tests ran in *".
It basically ensures that the test finished, and that there were e.g. no warnings.

Reply:
Also I'd include `*! KeyboardInterrupt !*`.